### ISSUE TITLE
Error executing 'postInstallation': Cannot read property 'trim' of undefined

### ISSUE THREAD

**Issue by senzacionale** [bitnami/bitnami-docker-sonarqube#3]
```

Welcome to the Bitnami sonarqube container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-sonarqube
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-sonarqube/issues


nami    INFO  Initializing postgresql-client
nami    INFO  postgresql-client successfully initialized
nami    INFO  Initializing sonarqube
sonarqu INFO  ==> Creating server folders...
sonarqu INFO  ==> Configuring database parameters...
sonarqu INFO  ==> Configuring server parameters...
sonarqu INFO  ==> Configure permissions...
sonarqu INFO  ==> Initializing SonarQube from scratch...
postgre INFO  Trying to connect to PostgreSQL server
postgre INFO  Found PostgreSQL server listening at postgresql:5432
postgre INFO  PostgreSQL server listening and working at postgresql:5432
sonarqu INFO  ==> Starting app...
sonarqu INFO  ==> Configuring username...
sonarqu INFO  ==> Configuring salt and password...
sonarqu INFO  ==> Enabling authentication...
Error executing 'postInstallation': Cannot read property 'trim' of undefined
```
any ideas why?

**Comment by andresbono** [comment]
Hi @senzacionale, how are you running the container? If you could share with us the specific `docker-compose.yml` file or `docker run` command that you are using, that would be great.

**Comment by andresbono** [comment]
Hi @senzacionale, we can't reproduce the issue. Could you copy & paste the command or specific `docker-compose.yml` file that you are using? That way we may be able to reproduce it.

**Comment by nazim-basha-vwcredit** [comment]
I am getting the same error too.

**Comment by marcosbc** [comment]
Hi @nazim-basha-vwcredit, we're still unable to reproduce this issue. We are using this `docker-compose.yml`:

```
version: '2'

services:
  postgresql:
    image: 'bitnami/postgresql:10'
    environment:
      - POSTGRESQL_USERNAME=bn_sonarqube
      - POSTGRESQL_DATABASE=bitnami_sonarqube
      - POSTGRESQL_PASSWORD=bitnami1234
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'postgresql_data:/bitnami'
  sonarqube:
    image: bitnami/sonarqube:latest
    ports:
      - '80:9000'
    environment:
      - POSTGRESQL_HOST=postgresql
      - SONARQUBE_DATABASE_NAME=bitnami_sonarqube
      - SONARQUBE_DATABASE_USER=bn_sonarqube
      - SONARQUBE_DATABASE_PASSWORD=bitnami1234
    volumes:
      - sonarqube_data:/bitnami
volumes:
  sonarqube_data:
    driver: local
  postgresql_data:
    driver: local
```

Then we run `docker-compose up -d`, `docker logs -f` and see everything initialize properly.

Is there anything we're missing?

**Comment by carrodher** [comment]
Hi, I am not able to reproduce the issue on my side, if I run `docker-compose up -d` everything works fine.
Can you paste here the output of the following command? In this way we can check the image version you are using when you run `docker-compose`
```bash
docker images | grep sonarqube && docker images |  grep postgre
```
In the same way, can you use the `NAMI_LOG_LEVEL=trace` environment variable and paste here the output surrounding the `sonarqu INFO  ==> Enabling authentication...` sentence?

**Comment by nazim-basha-vwcredit** [comment]
@marcosbc, @carrodher I am using AWS ECS to launch the docker instance in a private subnet and it connects to the postgres database running in another private subnet, which works fine. During the docker initialization, it registers its IP address and port number with the NLB, which is running in the public subnet. Due to some routing issues between NLB (public) and Docker instance (private), the NLB couldn't reach the docker instances and it marks the instances as unhealthy. Rerunning the same instance again, results into the mentioned error. Here is the captured logs:

2019-02-04 09:32:45Error executing 'postInstallation': Cannot read property 'trim' of undefined
2019-02-04 09:32:45sonarqu INFO ==> Enabling authentication...
2019-02-04 09:32:45sonarqu INFO ==> Configuring salt and password...
2019-02-04 09:32:45sonarqu INFO ==> Configuring username...
2019-02-04 09:31:43sonarqu INFO ==> Starting app...
2019-02-04 09:31:43postgre INFO PostgreSQL server listening and working at xxxxxx.us-east-1.rds.amazonaws.com:5432
2019-02-04 09:31:43postgre INFO Found PostgreSQL server listening at sonarqube-xxxx.us-east-1.rds.amazonaws.com:5432
2019-02-04 09:31:43sonarqu INFO ==> Initializing SonarQube from scratch...
2019-02-04 09:31:43postgre INFO Trying to connect to PostgreSQL server
2019-02-04 09:31:42sonarqu INFO ==> Configure permissions...
2019-02-04 09:31:42sonarqu INFO ==> Configuring server parameters...

**Comment by marcosbc** [comment]
Hi @nazim-basha-vwcredit, we're still unable to reproduce it but we have found something that we believe may be causing this issue.

We'll look into changing it, and let you know of any progress.

**Comment by marcosbc** [comment]
Hi @KacperLegowski, unfortunately we have not started working on this yet.

Do you keep getting this issue repeatedly? Do you also get the same error when initializing with a clean volume?

**Comment by fauzan16** [comment]
i got the same issue :

    [0m
[0m[1mWelcome to the Bitnami sonarqube container[0m
[0mSubscribe to project updates by watching [1mhttps://github.com/bitnami/bitnami-docker-sonarqube[0m
[0mSubmit issues and feature requests at [1mhttps://github.com/bitnami/bitnami-docker-sonarqube/issues[0m
[0m
nami    INFO  Initializing postgresql-client
nami    INFO  postgresql-client successfully initialized
nami    INFO  Initializing sonarqube
sonarqu DEBUG ==> Creating SonarQube System User...
sonarqu INFO  ==> Creating server folders...
sonarqu INFO  ==> Configuring database parameters...
sonarqu INFO  ==> Configuring server parameters...
sonarqu INFO  ==> Configure permissions...
sonarqu TRACE [configurePermissions] File to chown: /opt/bitnami/sonarqube/data
sonarqu TRACE [configurePermissions] File to chown: /opt/bitnami/sonarqube/temp
sonarqu TRACE [configurePermissions] File to chown: /opt/bitnami/sonarqube/logs
sonarqu TRACE [configurePermissions] File to chown: /opt/bitnami/sonarqube/extensions
sonarqu TRACE [configurePermissions] File to chown: /opt/bitnami/sonarqube/pids
sonarqu INFO  ==> Initializing SonarQube from scratch...
postgre INFO  Trying to connect to PostgreSQL server
postgre INFO  Found PostgreSQL server listening at sonar-postgres:5432
postgre TRACE [canConnect] Testing connection with 'sonar' user
postgre TRACE [showDatabases] Getting databases for 'sonar' user
postgre TRACE [execute] Executing psql with parameters -t,-p5432,-hsonar-postgres,-dpostgres,-Usonar,-cSELECT datname FROM pg_database;
postgre TRACE [showDatabases] Databases: " postgres\n sonar\n template1\n template0\n\n"
postgre TRACE [canConnect] Connection with 'sonar' user is successful
postgre INFO  PostgreSQL server listening and working at sonar-postgres:5432
sonarqu INFO  ==> Starting app...
sonarqu TRACE [runProgram] Executing: null su - sonarqube -c 'PATH=/opt/bitnami/java/bin:/opt/bitnami/postgresql/bin:/opt/bitnami/java/bin:/opt/bitnami/sonarqube/bin/linux-x86-64/:/opt/bitnami/nami/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /opt/bitnami/sonarqube/bin/linux-x86-64/sonar.sh start'
sonarqu TRACE code: 0
sonarqu TRACE stdout:

sonarqu TRACE stderr:

**Comment by fauzan16** [comment]
My temporary solution was downgrading the docker image to the version 7.5.0

**Comment by andresbono** [comment]
Thanks for letting us know, @fauzan16.

**Comment by kabo** [comment]
I'm getting the same error as x-fauzan-x describes (https://github.com/bitnami/bitnami-docker-sonarqube/issues/3#issuecomment-476152317).

```
docker run -p 9000:9000 \
  -e POSTGRESQL_HOST=xxx.cluster-xxx.eu-west-1.rds.amazonaws.com \
  -e SONARQUBE_DATABASE_USER=sonar \
  -e SONARQUBE_DATABASE_NAME=sonar \
  -e SONARQUBE_DATABASE_PASSWORD=xxx \
  -e NAMI_LOG_LEVEL=trace \
  bitnami/sonarqube:latest
```

I get the same error every time. I was able to create the database, but once it tries to start the app I just get exit code 0, no stdout and no stderr.

**Comment by andresbono** [comment]
Hi @kabo, was your issue solved when you applied https://github.com/bitnami/bitnami-docker-sonarqube/issues/11#issuecomment-517555407?

**Comment by kabo** [comment]
No, this is the new error I get after getting past #11

**Comment by fanefknroo** [comment]
check my sql below ... trim complains as it could not find the value on id=1.  You just put a result in its path and it will start up

`
bitnami_sonarqube=> SELECT created_at FROM properties WHERE id=1;
 created_at
------------
(0 rows)

bitnami_sonarqube=> SELECT * FROM properties WHERE id=1;
 id | prop_key | resource_id | user_id | is_empty | text_value | clob_value | created_at
----+----------+-------------+---------+----------+------------+------------+------------
(0 rows)

bitnami_sonarqube=> SELECT * FROM properties WHERE id=1;
 id | prop_key | resource_id | user_id | is_empty | text_value | clob_value | created_at
----+----------+-------------+---------+----------+------------+------------+------------
(0 rows)

bitnami_sonarqube=>
bitnami_sonarqube=> SELECT * FROM properties;
 id  |               prop_key               | resource_id | user_id | is_empty |                                   text_value                                    | clob_value |  created_at
-----+--------------------------------------+-------------+---------+----------+---------------------------------------------------------------------------------+------------+---------------
   2 | sonar.core.id                        |             |         | f        | xxxxxxxxxx                                                   |            | 1565075045312
   3 | sonar.forceAuthentication            |             |         | f        | true                                                                            |            | 1565075045256
  11 | sonar.cpd.exclusions                 |        4358 |         | f        | build/**                                                                        |            | 1568229894225
 181 | sonar.core.startTime                 |             |         | f        | 2019-09-20T11:02:51+0000                                                        |            | 1568977371787
  15 | sonar.coverage.exclusions            |             |         | f        | build/**,target/**                                                              |            | 1568230082255
bitnami_sonarqube=> SELECT * FROM properties WHERE id=1;
 id | prop_key | resource_id | user_id | is_empty | text_value | clob_value | created_at
----+----------+-------------+---------+----------+------------+------------+------------
(0 rows)

bitnami_sonarqube=> update properties set id=1 WHERE id=2;
UPDATE 1
bitnami_sonarqube=> SELECT created_at FROM properties WHERE id=1;
  created_at
---------------
 1565075045312
(1 row)
`

**Comment by javsalgar** [comment]
It seems to me that the error occurs because of a database that was not properly initialized. In the comments above I see that you were using an external database. Did you confirm that the database bitnami_sonarqube was created prior to launching the container?

**Comment by kabo** [comment]
@javsalgar For me the issue was adressed in #9

**Comment by javsalgar** [comment]
Thanks for letting us know @kabo :)

**Comment by bkrahmer** [comment]
> ## bitnami_sonarqube=> update properties set id=1 WHERE id=2;
> UPDATE 1

WOW!  That is a super-duper hacky workaround.  Much appreciated!  As a developer, I would like to point out that the database design is incorrect.  I would avoid using serial numbers at all, and if I did have to use them, I would never query by them.  The select queries should be using the prop_keys, not the ids!

**Comment by marcosbc** [comment]
Hi all, we just released [`SonarQube 9.0.0-10-debian-10-r0`](https://github.com/bitnami/bitnami-docker-sonarqube#900-debian-10-r0) which includes major changes, and the source code is now available in the [`rootfs` folder](https://github.com/bitnami/bitnami-docker-sonarqube/tree/master/9/debian-10/rootfs) in this repo.

As part of these changes, the Nami tool is no longer used for the bootstrapping so this should not be an issue anymore. Therefore I'm proceeding to close this issue.
